plugin/src/main/java/io/jenkins/plugins/redis/RedisFingerprintCleanup.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020, Jenkins project contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.plugins.redis;
+
+import hudson.Extension;
+import hudson.model.Fingerprint;
+import hudson.model.TaskListener;
+import jenkins.fingerprints.FingerprintCleanupThread;
+import redis.clients.jedis.ScanResult;
+import redis.clients.jedis.exceptions.JedisException;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension
+public class RedisFingerprintCleanup extends FingerprintCleanupThread {
+
+    private static final Logger LOGGER = Logger.getLogger(RedisFingerprintCleanup.class.getName());
+
+    public void execute(TaskListener listener) {
+        String currentPointer = redis.clients.jedis.ScanParams.SCAN_POINTER_START;
+
+        try {
+            do {
+                ScanResult<String> scanResult = RedisFingerprintStorage.get().getFingerprintIdsForCleanup(currentPointer);
+                List<String> fingerprintIds = scanResult.getResult();
+
+                try {
+                    List<Fingerprint> fingerprints = RedisFingerprintStorage.get().bulkLoad(fingerprintIds);
+                    for (Fingerprint fingerprint : fingerprints) {
+                        if (fingerprint != null) {
+                            cleanFingerprint(fingerprint, listener);
+                        }
+                    }
+                } catch (IOException e) {
+                    LOGGER.log(Level.WARNING, "Fingerprints found were malformed.", e);
+                }
+
+                currentPointer = scanResult.getCursor();
+            } while (!currentPointer.equals(redis.clients.jedis.ScanParams.SCAN_POINTER_START));
+        } catch (JedisException e) {
+            LOGGER.log(Level.WARNING, "Jedis failed to clean fingerprints. ", e);
```

Comment:
I wonder if you want to place this exception in the inner loop, so you can at least clean the other fingerprints.

Reply:
JedisException occurs during failure to connect with the database. And since we do a bulk load, I am not sure if we can actually retrieve the others

Reply:
https://github.com/jenkinsci/redis-fingerprint-storage-plugin/pull/23/files#diff-5a1ae3679c7223b516c0e27d1e4ca65fR168

Reply:
I plan to make that value `100` configurable soon. Didnt exactly get you, i was saying that we make one call to fetch some 100 fingerprint ids, and then one call to bulk load all of them together. The IOException is within the inner loop because that works iteratively.

Reply:
Yes, you are correct! thanks for pointing it out

Reply:
So IIUC, here is the issue: if the call fails that means I don't get the next cursor value. Hence i can't iterate any further. I may get stuck in a loop with the same cursor

Reply:
Gotcha; it would be cool if we bubble this up to users via UI warnings. I have to dig up the Jenkins feature that shows the dismiss-able warnings: https://www.bogotobogo.com/DevOps/Jenkins/images/Configuration/Manage_Jenkins.png